tests/test_core.py

```diff
@@ -775,3 +775,27 @@ def test_cross_kind_promotions(x, y):
         y = ndx.asarray(y)
     onnx_result = x + y
     np.testing.assert_equal(onnx_result.to_numpy(), np_result)
+
+
+# np.ma.allequal doesn't work correctly with identical objects, so we use a lambda to create distinct objects
+# https://github.com/numpy/numpy/issues/27201
+@pytest.mark.parametrize(
+    "x",
+    [
+        lambda: ndx.asarray([1, 2, 3], dtype=ndx.int64),
```

Comment:
It is okay to create such small arrays directly in the file scope. After all, the `lambda` has to live somewhere, too.

Reply:
That doesn't work because `x.copy()` doesn't make a deep copy